src/pyhf/tensor/pytorch_backend.py

```diff
@@ -118,6 +127,10 @@ def astensor(self, tensor_in, dtype='float'):
             tensor.shape[0]
         except IndexError:
             tensor = tensor.expand(1)
+
+        if self._mode == "GPU":
+            if torch.cuda.is_available():
+                tensor.cuda()
```

Comment:
It looks like this line doesn't actually do anything given that [`torch.Tensor.cuda()`](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.cuda)

> Returns a copy of this object in CUDA memory. If this object is already in CUDA memory and on the correct device, then no copy is performed and the original object is returned.

As an example
```python
import torch
example = torch.rand(3,3)

print(example)
print("First just call it")
if torch.cuda.is_available():
  example.cuda()
  print(example)

print("Now return it")
if torch.cuda.is_available():
  example = example.cuda()
  print(example)
```
returns (something like)
```
tensor([[0.7907, 0.6151, 0.5249],
        [0.9019, 0.4719, 0.4160],
        [0.8475, 0.7123, 0.6323]])
First just call it
tensor([[0.7907, 0.6151, 0.5249],
        [0.9019, 0.4719, 0.4160],
        [0.8475, 0.7123, 0.6323]])
Now return it
tensor([[0.7907, 0.6151, 0.5249],
        [0.9019, 0.4719, 0.4160],
        [0.8475, 0.7123, 0.6323]], device='cuda:0')
```

Reply:
Updated!